x-pack/auditbeat/magefile.go

```diff
@@ -79,185 +75,93 @@ func TestPackages() error {
-		},
-	}
+// Check runs fmt and update then returns an error if any modifications are found.
+func Check() {
+	mg.SerialDeps(mage.Format, Update, mage.Check)
+}
 
-	for _, args := range mage.Packages {
-		pkgType := args.Types[0]
-		switch pkgType {
-		case mage.TarGz, mage.Zip:
-			args.Spec.ReplaceFile("{{.BeatName}}.yml", shortConfig)
-			args.Spec.ReplaceFile("{{.BeatName}}.reference.yml", referenceConfig)
-		case mage.Deb, mage.RPM, mage.DMG:
-			args.Spec.ReplaceFile("/etc/{{.BeatName}}/{{.BeatName}}.yml", shortConfig)
-			args.Spec.ReplaceFile("/etc/{{.BeatName}}/{{.BeatName}}.reference.yml", referenceConfig)
-		default:
-			panic(errors.Errorf("unhandled package type: %v", pkgType))
-		}
-		if args.OS == "linux" {
-			rulesDest := archiveRulesDir
-			if pkgType != mage.TarGz {
-				rulesDest = linuxPkgRulesDir
-			}
-			args.Spec.Files[rulesDest] = sampleRules
-		}
-	}
+// IntegTest executes integration tests (it uses Docker to run the tests).
+func IntegTest() {
+	mage.AddIntegTestUsage()
+	defer mage.StopIntegTestEnv()
+	mg.SerialDeps(GoIntegTest, PythonIntegTest)
 }
 
-func generateReferenceConfig(spec mage.PackageSpec) error {
-	params := map[string]interface{}{
-		"Reference": true,
-		"ArchBits":  archBits,
-	}
-	return spec.ExpandFile(referenceConfigTemplate,
-		"{{.PackageDir}}/auditbeat.reference.yml", params)
+// UnitTest executes the unit tests.
+func UnitTest() {
+	mg.SerialDeps(GoUnitTest, PythonUnitTest)
 }
 
-func generateShortConfig(spec mage.PackageSpec) error {
-	params := map[string]interface{}{
-		"Reference": false,
-		"ArchBits":  archBits,
-	}
-	return spec.ExpandFile(shortConfigTemplate,
-		"{{.PackageDir}}/auditbeat.yml", params)
+// GoUnitTest executes the Go unit tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoUnitTest(ctx context.Context) error {
+	return mage.GoTest(ctx, mage.DefaultGoTestUnitArgs())
 }
 
-// archBits returns the number of bit width of the GOARCH architecture value.
-// This function is used by the auditd module configuration templates to
-// generate architecture specific audit rules.
-func archBits(goarch string) int {
-	switch goarch {
-	case "386", "arm":
-		return 32
-	default:
-		return 64
-	}
+// GoIntegTest executes the Go integration tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoIntegTest(ctx context.Context) error {
+	return mage.RunIntegTest("goIntegTest", func() error {
+		return mage.GoTest(ctx, mage.DefaultGoTestIntegrationArgs())
+	})
 }
 
-// Configs generates the auditbeat.yml and auditbeat.reference.yml config files.
-// Set DEV_OS and DEV_ARCH to change the target host for the generated configs.
-// Defaults to linux/amd64.
-func Configs() {
-	mg.Deps(makeConfigTemplates)
+// PythonUnitTest executes the python system tests.
+func PythonUnitTest() error {
+	mg.Deps(mage.BuildSystemTestBinary)
```

Comment:
I think `PythonUnitTest` should depend on `Fields`. Otherwise, it fails after a `mage clean`.

Reply:
I’ll fix that.

Reply:
Fixed.